src/v/kafka/protocol/schemata/generator.py

```diff
@@ -706,6 +770,18 @@ def decoder(self):
         assert plain_decoder[1]
         return plain_decoder[1], named_type
 
+    @property
+    def is_sensitive(self):
+        d = sensitive_map
+        for p in self._path:
+            d = d.get(p, None)
+            if d is None:
+                break
+        if not (d is None or d is True):
+            # This field likely matches a sub-path of a field in `sensitive_map`.
+            return False
+        return d
```

Comment:
This was adjusted because the original assertion didn't handle ancestors of sensitive fields well.

```
...
  File "<template>", line 145, in top-level template code
  File "/home/andrew/.pex/installed_wheels/6088930bfe239f0e6710546ab9c19c9ef35e29792895fed6e6e31a023a182a61/Jinja2-3.1.2-py3-none-any.whl/jinja2/environment.py", line 485, in getattr
    return getattr(obj, attribute)
  File "/home/andrew/Repos/redpanda/src/v/kafka/protocol/schemata/generator.py", line 785, in is_sensitive
    assert d is None or d is True, \
AssertionError: expected field 'Configs' to be missing or True; field path: ('AlterConfigsRequestData', 'Resources', 'Configs'), remaining path: {'Value': True}
[37/198] Running kafka request codegen on init_producer_id_response.json
```

I've updated the code to make this more obvious.